app/src/main/java/com/gaurav/avnc/ui/vnc/FrameView.kt

```diff
@@ -77,6 +84,13 @@ class FrameView(context: Context?, attrs: AttributeSet? = null) : GLSurfaceView(
         frameState.setViewportSize(w.toFloat(), h.toFloat())
     }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onResolvePointerIcon(e: MotionEvent, pointerIndex: Int): PointerIcon {
+        if (!prefs.input.hideLocalCursor)
+            return PointerIcon.getSystemIcon(context, PointerIcon.TYPE_DEFAULT)
+
+        return PointerIcon.getSystemIcon(context, PointerIcon.TYPE_NULL)
+    }
```

Comment:
Instead of overriding `onResolvePointerIcon()`, we should use [setPointerIcon()](https://developer.android.com/reference/android/view/View#setPointerIcon(android.view.PointerIcon)) to hide the pointer. We can update the pointer at the end of [initialize()](https://github.com/gujjwal00/avnc/blob/0188996bc0b126d4dcb727821b6a6fa1dd7ede62/app/src/main/java/com/gaurav/avnc/ui/vnc/FrameView.kt#L73), like this:
```kotlin
        if (Build.VERSION.SDK_INT >= 24 && viewModel.pref.input.hideLocalCursor)
            pointerIcon = PointerIcon.getSystemIcon(context, PointerIcon.TYPE_NULL)
```

Overriding `onResolvePointerIcon()` like this has some drawbacks:
- We are doing a map-lookup via `prefs.input.hideLocalCursor` for every event, even though the lookup result won't  change during the session.
- Its more verbose compared to `setPointerIcon()`, which suits our requirement perfectly.